src/Time/StepChoosers/StepChooser.hpp

```diff
@@ -60,20 +60,25 @@ class StepChooser : public PUP::able {
   /// The `last_step_magnitude` parameter describes the step size to be
   /// adjusted.  It may be the step size or the slab size, or may be
   /// infinite if the appropriate size cannot be determined.
+  ///
+  /// The return value of this function contains the desired step size
+  /// and a `bool` indicating whether the step should be accepted. If
+  /// the `bool` is `false`, the current time step will be recomputed with
+  /// a strictly smaller step, guided by the requested time step value.
   template <typename Metavariables, typename DbTags>
-  double desired_step(
+  std::pair<double, bool> desired_step(
```

Comment:
Would it be reasonable to make the return value here a `std::optional<double>`, meaning either "do a step with this size" or "don't do a step"? So the question is: Will the "desired step" `double` be used also when the "acceptance" `bool` is `false`? If this is what you mean with "guided by the requested time step value", please clarify, e.g. by rewording to "... the desired step value returned by this function".

Reply:
The return value is used when the step is rejected to choose the step size when retrying the step; it is assumed that when the step is rejected the returned step size is strictly smaller than the previously tried step. I'll clarify the documentation.